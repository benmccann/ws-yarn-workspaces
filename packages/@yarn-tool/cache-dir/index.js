"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getYarnCacheDir = void 0;
const upath2_1 = require("upath2");
const fs_extra_1 = require("fs-extra");
const cross_spawn_extra_1 = require("cross-spawn-extra");
function getYarnCacheDir() {
    try {
        let cp = cross_spawn_extra_1.sync('yarn', [
            'config',
            'current',
            '--json',
        ], {
            stripAnsi: true,
            env: process.env,
        });
        let data = JSON.parse(JSON.parse(cp.stdout.toString()).data);
        if (data.tempFolder) {
            return upath2_1.normalize(data.tempFolder);
        }
    }
    catch (e) {
    }
    if (process.env.YARN_CACHE_FOLDER && fs_extra_1.pathExistsSync(process.env.YARN_CACHE_FOLDER)) {
        return upath2_1.normalize(process.env.YARN_CACHE_FOLDER);
    }
}
exports.getYarnCacheDir = getYarnCacheDir;
exports.default = getYarnCacheDir;
//# sourceMappingURL=index.js.map