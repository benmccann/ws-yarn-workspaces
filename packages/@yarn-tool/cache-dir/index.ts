import { join, normalize } from "upath2";
import { pathExistsSync } from 'fs-extra';
import { sync as crossSpawnExtra } from 'cross-spawn-extra';

export function getYarnCacheDir(): string
{
	try
	{
		let cp = crossSpawnExtra('yarn', [
			'config',
			'current',
			'--json',
		], {
			stripAnsi: true,
			env: process.env,
		});

		let data = JSON.parse(JSON.parse(cp.stdout.toString()).data);

		if (data.tempFolder)
		{
			return normalize(data.tempFolder)
		}
	}
	catch (e)
	{

	}

	if (process.env.YARN_CACHE_FOLDER && pathExistsSync(process.env.YARN_CACHE_FOLDER))
	{
		return normalize(process.env.YARN_CACHE_FOLDER)
	}
}

export default getYarnCacheDir
