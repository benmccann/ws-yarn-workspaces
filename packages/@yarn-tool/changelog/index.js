"use strict";
/**
 * Created by user on 2020/6/15.
 */
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
tslib_1.__exportStar(require("./lib/types"), exports);
tslib_1.__exportStar(require("./lib/lerna"), exports);
tslib_1.__exportStar(require("./lib/yargs-setting"), exports);
const lerna_1 = require("./lib/lerna");
exports.default = lerna_1.updateChangelogByCwd;
//# sourceMappingURL=index.js.map