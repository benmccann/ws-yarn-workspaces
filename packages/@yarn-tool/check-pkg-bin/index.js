"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const pkg_1 = require("./lib/pkg");
(0, tslib_1.__exportStar)(require("./lib/core"), exports);
(0, tslib_1.__exportStar)(require("./lib/ws"), exports);
(0, tslib_1.__exportStar)(require("./lib/pkg"), exports);
exports.default = pkg_1.checkPkgDir;
//# sourceMappingURL=index.js.map