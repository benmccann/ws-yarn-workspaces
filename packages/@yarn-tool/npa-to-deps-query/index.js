"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.queryDepsValueByNpa = exports.queryDepsValueByNpaResult = void 0;
const tslib_1 = require("tslib");
const npa_to_deps_1 = tslib_1.__importDefault(require("@yarn-tool/npa-to-deps"));
const bluebird_1 = tslib_1.__importDefault(require("bluebird"));
const queryVersion_1 = require("@yarn-tool/pkg-version-query/lib/queryVersion");
function queryDepsValueByNpaResult(depsResult, options) {
    return bluebird_1.default.resolve(depsResult)
        .then(depsResult => {
        var _a, _b;
        const name = depsResult.name;
        const operator = (_a = depsResult.operator) !== null && _a !== void 0 ? _a : '';
        if (depsResult.fetchQuery) {
            return (0, queryVersion_1.queryVersionWithCache)(depsResult.name, (_b = depsResult.semver) !== null && _b !== void 0 ? _b : depsResult.result.fetchSpec, options === null || options === void 0 ? void 0 : options.queryOptions)
                .then(version => {
                return {
                    name,
                    value: `${operator}${version}`,
                };
            });
        }
        return {
            name,
            value: `${operator}${depsResult.semver}`,
        };
    });
}
exports.queryDepsValueByNpaResult = queryDepsValueByNpaResult;
function queryDepsValueByNpa(input, options) {
    return bluebird_1.default.resolve()
        .then(() => {
        return (0, npa_to_deps_1.default)(input, options);
    })
        .then((result) => {
        return queryDepsValueByNpaResult(result, options)
            .then((ret) => {
            return {
                ...result,
                ...ret,
            };
        });
    });
}
exports.queryDepsValueByNpa = queryDepsValueByNpa;
exports.default = queryDepsValueByNpa;
//# sourceMappingURL=index.js.map