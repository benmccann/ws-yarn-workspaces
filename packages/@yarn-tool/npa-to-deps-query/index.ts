import {
	npaToDepsValue,
	npaResultToDepsValue,
	IOptions as IOptionsDepsValue,
	IDepsResult,
} from '@yarn-tool/npa-to-deps';
import Bluebird from 'bluebird';
import { ITSResolvable } from 'ts-type';
import { queryVersionWithCache } from '@yarn-tool/pkg-version-query/lib/queryVersion';
import { IOptionsQueryVersion } from '@yarn-tool/pkg-version-query/lib/types';
import { Options } from 'package-json';

export interface IOptions extends IOptionsDepsValue
{
	queryOptions?: IOptionsQueryVersion<Options>,
}

export function queryDepsValueByNpaResult(depsResult: ITSResolvable<IDepsResult>, options?: IOptions)
{
	return Bluebird.resolve(depsResult)
		.then(depsResult =>
		{
			const name = depsResult.name;
			const operator = depsResult.operator ?? '';

			if (depsResult.fetchQuery)
			{
				return queryVersionWithCache(depsResult.name, depsResult.semver ?? depsResult.result.fetchSpec, options?.queryOptions)
					.then(version =>
					{
						return {
							name,
							value: `${operator}${version}`,
						};
					})
			}

			return {
				name,
				value: `${operator}${depsResult.semver}`,
			};
		})
}

export function queryDepsValueByNpa(input: string, options?: IOptions)
{
	return Bluebird.resolve()
		.then(() =>
		{
			return npaToDepsValue(input, options)
		})
		.then((result) =>
		{
			return queryDepsValueByNpaResult(result, options)
				.then((ret) => {
					return {
						...result,
						...ret,
					}
				})
		})
}

export default queryDepsValueByNpa
