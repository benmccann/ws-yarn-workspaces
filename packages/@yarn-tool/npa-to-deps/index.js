"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.npaToDepsValue = exports.npaResultToDepsValue = void 0;
const npm_package_arg_util_1 = require("@yarn-tool/npm-package-arg-util");
const parseSimpleSemVerRange_1 = require("@lazy-node/semver-simple-parse/lib/parseSimpleSemVerRange");
function npaResultToDepsValue(result, options) {
    var _a, _b, _c;
    let semver;
    let operator;
    let fetchQuery;
    switch (result.type) {
        case 'git':
            semver = result.saveSpec;
            break;
        case 'tag':
            if (((_a = result.fetchSpec) === null || _a === void 0 ? void 0 : _a.length) && (options === null || options === void 0 ? void 0 : options.preserveTag)) {
                semver = result.fetchSpec;
            }
            else {
                operator = '^';
                fetchQuery = true;
            }
            break;
        case 'range':
            if ((_b = result.fetchSpec) === null || _b === void 0 ? void 0 : _b.length) {
                let ls = (0, parseSimpleSemVerRange_1.parseSimpleSemVerRange)(result.fetchSpec);
                if (ls.length > 1) {
                    semver = result.fetchSpec;
                }
                else if (options === null || options === void 0 ? void 0 : options.preserveRange) {
                    semver = result.fetchSpec;
                }
                else if (ls.length === 1) {
                    let entry = ls[0];
                    if (!entry.operator) {
                        semver = result.fetchSpec;
                        fetchQuery = true;
                    }
                    else if (entry.operator !== '^') {
                        semver = result.fetchSpec;
                    }
                    else {
                        operator = '^';
                        fetchQuery = true;
                    }
                }
                else {
                    operator = '^';
                    fetchQuery = true;
                }
            }
            else {
                operator = '^';
                fetchQuery = true;
            }
            break;
        default:
            if ((_c = result.fetchSpec) === null || _c === void 0 ? void 0 : _c.length) {
                semver = result.fetchSpec;
            }
            break;
    }
    return {
        name: result.name,
        semver,
        operator,
        fetchQuery,
        result,
    };
}
exports.npaResultToDepsValue = npaResultToDepsValue;
function npaToDepsValue(arg, options) {
    let result = (0, npm_package_arg_util_1.npa)(arg, options === null || options === void 0 ? void 0 : options.where);
    return npaResultToDepsValue(result);
}
exports.npaToDepsValue = npaToDepsValue;
exports.default = npaToDepsValue;
//# sourceMappingURL=index.js.map