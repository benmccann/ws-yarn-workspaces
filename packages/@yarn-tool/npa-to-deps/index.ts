import { IResult, npa } from '@yarn-tool/npm-package-arg-util';
import { parseSimpleSemVerRange } from '@lazy-node/semver-simple-parse/lib/parseSimpleSemVerRange';

export interface IOptions
{
	where?: string,
	/**
	 * when input is tag, preserve it
	 */
	preserveTag?: boolean,

	/**
	 * when input is range, preserve it
	 */
	preserveRange?: boolean,
}

export interface IDepsResult<T extends IResult = IResult>
{
    name: string;
    semver?: string;
    operator?: string;
    fetchQuery?: boolean;
    result: T;
}

export function npaResultToDepsValue<T extends IResult>(result: T, options?: IOptions): IDepsResult<T>
{
	let semver: string;
	let operator: string;
	let fetchQuery: boolean;

	switch (result.type)
	{
		case 'git':
			semver = result.saveSpec
			break;
		case 'tag':
			if (result.fetchSpec?.length && options?.preserveTag)
			{
				semver = result.fetchSpec
			}
			else
			{
				operator = '^';
				fetchQuery = true;
			}
			break;
		case 'range':

			if (result.fetchSpec?.length)
			{
				let ls = parseSimpleSemVerRange(result.fetchSpec);

				if (ls.length > 1)
				{
					semver = result.fetchSpec
				}
				else if (options?.preserveRange)
				{
					semver = result.fetchSpec
				}
				else if (ls.length === 1)
				{
					let entry = ls[0];

					if (!entry.operator)
					{
						semver = result.fetchSpec
						fetchQuery = true;
					}
					else if (entry.operator !== '^')
					{
						semver = result.fetchSpec
					}
					else
					{
						operator = '^';
						fetchQuery = true;
					}

				}
				else
				{
					operator = '^';
					fetchQuery = true;
				}
			}
			else
			{
				operator = '^';
				fetchQuery = true;
			}

			break;
		default:

			if (result.fetchSpec?.length)
			{
				semver = result.fetchSpec
			}

			break;
	}

	return {
		name: result.name,
		semver,
		operator,
		fetchQuery,
		result,
	}
}

export function npaToDepsValue<T extends IResult = IResult>(arg: string, options?: IOptions)
{
	let result = npa(arg, options?.where);

	return npaResultToDepsValue<T>(result as any)
}

export default npaToDepsValue
