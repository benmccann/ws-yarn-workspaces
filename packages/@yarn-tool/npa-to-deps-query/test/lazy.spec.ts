import { basename, extname } from 'path';
import { npaToDepsValue } from '@yarn-tool/npa-to-deps';
import { queryDepsValueByNpaResult } from '../index';

[
	'jest@12',
	'jest@>12',
	'jest@<12',
	'jest@<=12',
	'jest@>=12',
	'jest@^12',
].forEach((input) =>
{

	test(input, async () =>
	{
		let depsResult = npaToDepsValue(input);

		let actual = await queryDepsValueByNpaResult(depsResult);

		console.dir({
			depsResult,
			actual,
		});

		expect(actual).toHaveProperty('name', depsResult.name);

		const operator = depsResult.operator ?? '';

		if (operator === '^')
		{
			expect(actual.value).toMatch(/^\^\d+/);
		}
		else
		{
			expect(actual.value).toContain(`${operator}${depsResult.semver}`);
		}

		expect(actual).toMatchSnapshot();

	});

});
