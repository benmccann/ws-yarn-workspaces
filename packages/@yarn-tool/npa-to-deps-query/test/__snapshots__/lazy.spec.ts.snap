// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest@<=12 1`] = `
Object {
  "name": "jest",
  "value": "<=12",
}
`;

exports[`jest@<12 1`] = `
Object {
  "name": "jest",
  "value": "<12",
}
`;

exports[`jest@>=12 1`] = `
Object {
  "name": "jest",
  "value": ">=12",
}
`;

exports[`jest@>12 1`] = `
Object {
  "name": "jest",
  "value": ">12",
}
`;

exports[`jest@^12 1`] = `
Object {
  "name": "jest",
  "value": "^12.1.1",
}
`;

exports[`jest@12 1`] = `
Object {
  "name": "jest",
  "value": "12.1.1",
}
`;
